giraffe/src/transforms/band.ts

```diff
@@ -0,0 +1,210 @@
+    }
+    if (minMaxMap[line.lineName][MAX]) {
+      line[MAX] = {...minMaxMap[line.lineName][MAX], fill: line.fill}
+    }
+  })
+  return bands as Band[]
+}
+
+export const getBandIndexMap = (
+  fillColumnMap: ColumnGroupMap
+): BandIndexMap => {
+  const bandIndices = {
+    rowIndices: [],
+    minIndices: [],
+    maxIndices: [],
+  }
+
+  if (Array.isArray(fillColumnMap.mappings)) {
+    fillColumnMap.mappings.forEach((line, index) => {
+      if (line[RESULT] === MAX) {
+        bandIndices.maxIndices.push(index)
+      } else if (line[RESULT] === MIN) {
+        bandIndices.minIndices.push(index)
+      } else {
+        bandIndices.rowIndices.push(index)
+      }
+    })
+  }
+
+  return bandIndices
+}
+
+export const getBandLineIndices = (fillColumnMap: ColumnGroupMap): number[] => {
+  const bandLineIndices = Array.isArray(fillColumnMap.mappings)
+    ? fillColumnMap.mappings.reduce((indices, line, index) => {
+        if (line[RESULT] && line[RESULT] !== MAX && line[RESULT] !== MIN) {
+          indices.push(index)
+        }
+        return indices
+      }, [])
+    : []
+
+  return bandLineIndices as number[]
+}
+
+export const getBandName = (fillColumnMap: ColumnGroupMap): string => {
+  return Array.isArray(fillColumnMap.mappings)
+    ? fillColumnMap.mappings.reduce((name, line) => {
+        if (!name && line[RESULT] !== MAX && line[RESULT] !== MIN) {
+          name = line[RESULT]
+        }
+        return name
+      }, '')
+    : ''
+}
+
+export const mapCumulativeValuesToTimeRange = (
+  timesCol: NumericColumnData,
+  valuesCol: NumericColumnData,
+  fillCol: NumericColumnData
+): CumulativeValuesByTime => {
+  const cumulativeValues: CumulativeValuesByTime = {}
+
+  timesCol.forEach((time, index) => {
+    if (!cumulativeValues[time]) {
+      cumulativeValues[time] = {}
+    }
+    const maxGroup = fillCol[index]
+
+    if (!cumulativeValues[time][maxGroup - 1]) {
+      for (let i = 0; i < maxGroup; i += 1) {
+        if (!cumulativeValues[time][i]) {
+          cumulativeValues[time][i] = 0
+        }
+      }
+    }
+    cumulativeValues[time][maxGroup] =
+      maxGroup - 1 >= 0
+        ? cumulativeValues[time][maxGroup - 1] + valuesCol[index]
+        : valuesCol[index]
+  })
+  return cumulativeValues
+}
+
+export const bandTransform = (
+  inputTable: Table,
+  xColumnKey: string,
+  yColumnKey: string,
+  fillColKeys: string[],
+  colors: string[]
+): BandLayerSpec => {
+  const [fillColumn, fillColumnMap] = createGroupIDColumn(
+    inputTable,
+    fillColKeys
+  )
+
+  const table = inputTable.addColumn(FILL, 'number', fillColumn)
+  const xCol = table.getColumn(xColumnKey, 'number')
+  const yCol = table.getColumn(yColumnKey, 'number')
+  const fillScale = getNominalColorScale(fillColumnMap, colors)
+  const bandFillColors = []
+  const lineData: LineData = {}
+
+  let xMin = Infinity
+  let xMax = -Infinity
+  let yMin = Infinity
+  let yMax = -Infinity
```

Comment:
these seem backwards - i'd expect `xMin` to be `-Infinity` and `xMax` to `Infinity`

Reply:
These are backwards because they get set to a different value. For example, the first time is pretty guaranteed to set a new xMin because nothing can be greater than Infinity. Thus xMin gets set and will be set again if the next value is less than the new value.